assets/js/frontend/course-video/youtube-extension.js

```diff
@@ -0,0 +1,78 @@
+( () => {
+	const {
+		courseVideoRequired,
+		courseVideoAutoComplete,
+		courseVideoAutoPause,
+	} = window.sensei.courseVideoSettings;
+
+	const handleVisibilityChange = ( player ) => () => {
+		if ( document.hidden ) {
+			player.pauseVideo();
+		}
+	};
+
+	const preventClick = ( event ) => {
+		event.preventDefault();
+		return false;
+	};
+
+	const onYouTubePlayerStateChange = ( event ) => {
+		const playerStatus = event.data;
+
+		if ( courseVideoRequired && playerStatus === YT.PlayerState.ENDED ) {
+			document
+				.querySelectorAll( '[data-id="complete-lesson-button"]' )
+				.forEach( ( button ) => {
+					button.removeEventListener( 'click', preventClick );
+					button.disabled = false;
+				} );
+		}
+
+		if (
+			courseVideoAutoComplete &&
+			playerStatus === YT.PlayerState.ENDED
+		) {
+			// submit complete lesson form
+			const form = document.querySelector(
+				'[data-id="complete-lesson-form"]'
+			);
+			if ( form ) {
+				form.dispatchEvent( new Event( 'submit' ) );
+			}
+		}
+	};
+
+	const initPlayer = ( iframe ) => {
+		const player = new YT.Player( iframe, {
+			events: {
+				onStateChange: onYouTubePlayerStateChange,
+			},
+		} );
+
+		if ( courseVideoAutoPause && document.hidden !== undefined ) {
+			// eslint-disable-next-line @wordpress/no-global-event-listener
+			document.addEventListener(
+				'visibilitychange',
+				handleVisibilityChange( player ),
+				false
+			);
+		}
+	};
+
+	window.onYouTubeIframeAPIReady = function () {
```

Comment:
Note that this could break other plugins integrating with the embed, since it overwrites any existing handler. Not sure if there is an event alternative to hook up using `addEventListener` instead, if not, it might be worth to check if a `onYouTubeIframeAPIReady` function already exists, and store and call it in the handler if it does.